crypto/src/dex/execution.rs

```diff
@@ -0,0 +1,42 @@
+use crate::asset;
+use crate::dex::lp::TradingFunction;
+use crate::dex::trading_pair::DirectedTradingPair;
+use anyhow::Result;
+
+pub struct Path {
+    pair: DirectedTradingPair,
+    route: Vec<asset::Id>,
+    phi: TradingFunction,
```

Comment:
nit: Not sure if we want to use the greek characters from the design docs in code; we avoided that in the implementation of swaps in favor of more descriptive property names.